### ISSUE TITLE
Modal doesn't disappear after close

### ISSUE THREAD

**Issue by dginsburg** [ideashower/ShareKit#24]
Share a URL
Select More...
Select Email
Press Cancel

The view isn't removed

**Comment by ideashower** [comment]
Not able to reproduce this.  Are you in the example app or your own implementation?

**Comment by dginsburg** [comment]
My own implementation. This happens with any of the services or actions. If I go through the more menu, then choose a service and cancel it, the view doesn't disappear.

**Comment by ideashower** [comment]
Did you subclass, hack, or override any of the core ShareKit classes?  I'd have to see an example to sort this out.

**Comment by dginsburg** [comment]
Yes. I changed the modalPresentationStyle

nav.modalPresentationStyle = UIModalPresentationFormSheet;

Removing that and using the default eliminates this problem but it's not the UI I want. I'll figure it out.

**Comment by ideashower** [comment]
Weird it would cause that, thanks for the details, I'll add this to the fix-it list.  If you discover a fix, let me know!  Thanks!

**Comment by dginsburg** [comment]
The weirdness seems to be around the animation. The problem goes away if I send NO for the animation of the modal present and dismiss. This is not the first time I've seen weirdness when stringing two animated events back to back. If you animate UIToolbar changes and the time periods overlap then the buttons won't always display correctly.

**Comment by dginsburg** [comment]
And once again I hack things with timers. Steve Jobs kills a puppy every time I do this. The code below in SHK.m seems to be working for my needs. I don't claim it's a good solution.

```
- (void)showPendingView
{
    if (pendingView)
        [self showViewController:pendingView];
}

- (void)viewWasDismissed
{
    self.isDismissingView = NO;

    if (currentView != nil)
        currentView = nil;

    if (pendingView)
    {
        [self performSelector:@selector(showPendingView) withObject:nil afterDelay:0.02];
//      [self showViewController:pendingView];
        return;
    }
}
```

**Comment by ideashower** [comment]
Merged into 0.2.0b1

**Comment by Nicrob64** [comment]
This fix can cause the sharer to reopen when it is closed via cancel. 
When sharing to twitter through [SHKTwitter shareItem:item]; if cancel is selected it will reopen the view after the 0.02s delay, leading to the user not being able to return to the app.
